packages/backend-defaults/src/entrypoints/rootConfig/createConfigSecretEnumerator.ts

```diff
@@ -14,18 +14,22 @@
  * limitations under the License.
  */
 
-import { LoggerService } from '@backstage/backend-plugin-api';
+import type {
+  AuditorService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
 import type { Config } from '@backstage/config';
 import { ConfigSchema, loadConfigSchema } from '@backstage/config-loader';
 import { getPackages } from '@manypkg/get-packages';
 
 /** @public */
-export async function createConfigSecretEnumerator(options: {
-  logger: LoggerService;
-  dir?: string;
-  schema?: ConfigSchema;
-}): Promise<(config: Config) => Iterable<string>> {
-  const { logger, dir = process.cwd() } = options;
+export async function createConfigSecretEnumerator(
+  options: {
+    dir?: string;
+    schema?: ConfigSchema;
+  } & ({ logger: LoggerService } | { auditor: AuditorService }),
```

Comment:
Aiming for this as the way forward for redactions outside of the LoggerService: https://github.com/backstage/backstage/pull/24730